main.go

```diff
@@ -121,50 +125,29 @@ func main() {
 		os.Exit(1)
 	}
 
-	vcnClient, err := core.NewVirtualNetworkClientWithConfigurationProvider(ociAuthConfigProvider)
+	clientProvider, err := scope.NewClientProvider(authConfig)
 	if err != nil {
-		setupLog.Error(err, "unable to create OCI VCN Client")
-		os.Exit(1)
-	}
-
-	lbClient, err := networkloadbalancer.NewNetworkLoadBalancerClientWithConfigurationProvider(ociAuthConfigProvider)
-	if err != nil {
-		setupLog.Error(err, "unable to create OCI LB Client")
-		os.Exit(1)
-	}
-
-	identityClient, err := identity.NewIdentityClientWithConfigurationProvider(ociAuthConfigProvider)
-	if err != nil {
-		setupLog.Error(err, "unable to create OCI Identity Client")
+		setupLog.Error(err, "unable to create OCI ClientProvider")
 		os.Exit(1)
 	}
 
 	if err = (&controllers.OCIClusterReconciler{
-		Client:                    mgr.GetClient(),
-		Scheme:                    mgr.GetScheme(),
-		VCNClient:                 vcnClient,
-		NetworkLoadBalancerClient: lbClient,
-		IdentityClient:            identityClient,
-		Region:                    region,
-		Recorder:                  mgr.GetEventRecorderFor("ocicluster-controller"),
+		Client:         mgr.GetClient(),
+		Scheme:         mgr.GetScheme(),
+		Region:         region,
```

Comment:
can you test with instance principal please? just want to make sure it works

Reply:
I have tested this with Instance Principals and it is working.

Reply:
I've tests in region and cross region clusters with Instance Principals. It is working as expected.